### ISSUE TITLE
Issues with Custom Alias: Player Name Recognition

### ISSUE THREAD

**Issue by VimaMT** [Zrips/CMI#8138]
### Description of Issue

This is a continuation of a topic I tried to open on July 29: https://github.com/Zrips/CMI/discussions/8025

I am setting up a custom alias that enhances the feed command. One check I have in the alias is supposed to prevent a player from feeding themselves:

if:parmMissing@! if:feedPermOther! check:$1==%cmi_user_name%~! asConsole! msg! [playerName] &b[&b&lCW&b] &cYou are not permitted to feed yourself with this command.

It works well as long as the player has entered more than two characters of their name. But if they only enter one or two characters of their name (ie, /feed Gs) .. it feeds them (GShum).

### Version Information

```shell
[19:46:05 INFO]: --------------------------------------------------
[19:46:05 INFO]: CMI: 9.6.3.0 BungeeCord CMIB  SqLite
[19:46:05 INFO]: CMILib: 1.4.1.2 
[19:46:05 INFO]: Server: Purpur 1.20.1-R0.1-SNAPSHOT
[19:46:05 INFO]: CMI economy: Disabled Vault: 1.7.3-CMI CMI Chat: Enabled 
[19:46:05 INFO]: Modules -> 44 enabled 15 disabled: hpBossBar, mirror, launchPad, tablist, armorstand, elytraExploit, moneyCheque, elytraBoost, elytraLaunch, bossBarCompass, playerCombat, votifier, sitAnimation, spawnerCharge, totemBehavior
[19:46:05 INFO]: --------------------------------------------------
```

### Errors

```shell
No Errors visible - The scripted checks just refuse to give the anticipated results.
```

### Relevant Config Sections

```yaml
Our CMI Configuration currently uses a Minimum Nickname length of 2. Changing this value does not seem to impact us. We have issues regardless of this setting.
```

### Agreements

- [X] My server is supported by CMI.
- [X] My version of CMI at the time of this report is up to date.
- [X] I have searched the github and asked around before making this report.

**Comment by Zrips** [comment]
Can you provide more of a full setup what you have? As it doesn't look like its being misbehaving when you enter none full names, atleast as of my testing. But you do have extra checks at the start which might cause this check being ignored

**Comment by VimaMT** [comment]
Looking at it more now. I just installed the new CMI with the name recognition fix in it, and will try it out. It always takes me a bit to get my head around these things after I've let them sit for a while. :-) Will give you more feedback today or tomorrow.

**Comment by VimaMT** [comment]
Ok, it seems to have resolved itself with that latest CMI release. Here's the object and scenario.

Object: Become more EULA compliant. Normal Players get /feed via CMI Perms. Donator Players get /feed <other> via CMI Perms.  We did not want to allow a player to buy /feed. We created a script that will do only one or the other depending upon the CMI permissions the player has.  Thus, a player with cmi.command.feed.others can feed other players, but not themselves. A player with cmi.command.feed (normal player) can feed themselves, but not others.

Scenario:
- I provided cmi.command.feed.others to a test player named GShum
- Gshum did NOT have cmi.command.feed
- GShum, thru the script, could still feed himself by using /feed GS
The comparison operation inside the script thought that GS was another player on the server and not GShum. He was not supposed to be able to do that.

This latest CMI appears, on the surface, to have fixed that. I want to play around with it another day and doubly-verify.
